trin-core/src/cli.rs

```diff
@@ -100,6 +100,16 @@ pub struct TrinConfig {
         help = "Maximum number of kilobytes of total data to store in the DB"
     )]
     pub kb: u32,
+
+    #[structopt(
+        long = "enable-metrics",
+        help = "Enable prometheus metrics reporting (requires --metrics-url)",
+        requires = "metrics-url"
+    )]
+    pub enable_metrics: bool,
+
+    #[structopt(long = "metrics-url", help = "URL for prometheus server")]
+    pub metrics_url: Option<String>,
```

Comment:
If I understand what you're suggesting... let's remove the `--enable-metrics` flag and implicitly "enable metrics" whenever a `--metrics-url` arg is provided - otherwise metrics are disabled... This makes sense and is cleaner... but I can't help but feel like there is some value to an explicit `--enable-metrics` flag. It would make sense if we have a default `metrics_url` (which admittedly we don't right now) so then you wouldn't need to use the `--metrics-url` flag, but this could be something we use down the line?

Hmm, I'm just thinking out loud there. Honestly, I'm 60/40 on keeping `--enable-metrics` - it's not that much overhead, I think explicit > implicit can be beneficial in some cases. We can always remove it down the line if it proves completely unnecessary. But I'm interested in your thoughts

Reply:
:+1: - will also add unit tests for these

Reply:
I don't think we'd want default metrics reporting, it seems to me like the metrics feature would be more useful in terms of users being able to run their own metrics locally, and us collecting metrics from our testnet nodes. I'm not sure some kind of default url that collects every users data is something I'd go for - data privacy and all that good stuff. 

I'm also not against `--enable-metrics-with-url`... https://github.com/ethereum/trin/issues/262